--- snarkjs-0.7.5/src/groth16_prove.js
+++ snarkjs-controlled/src/groth16_prove.js
@@ -100,8 +100,8 @@
     const buffBasesH = await binFileUtils.readSection(fdZKey, sectionsZKey, 9);
     const resH = await curve.G1.multiExpAffine(buffBasesH, buffPodd_T, logger, "multiexp H");
 
-    const r = curve.Fr.random();
-    const s = curve.Fr.random();
+    const r = curve.Fr.e(options.fixtureR);
+    const s = curve.Fr.e(options.fixtureS);
 
     proof.pi_a  = G1.add( proof.pi_a, zkey.vk_alpha_1 );
     proof.pi_a  = G1.add( proof.pi_a, G1.timesFr( zkey.vk_delta_1, r ));
